### DISCUSSION TITLE
Write up on my experience with Algolia + NextJS

### DISCUSSION THREAD

**Discussion by avremel** [algolia/instantsearch#5822] [Show and tell]
Over the last year, I've been working on implementing Algolia with NextJS for AjMadison.  I wrote an article on things I wish I had known before I started, in the hope that others can benefit from it.

https://www.avikaminetzky.dev/posts/algolia-ecommerce-nextjs

I started working on this project when React Instant Search Hooks was [still in beta](https://twitter.com/algolia/status/1483841672320532489?t=-rj9rDWwMnbLUnpXDsjmBQ&s=19) and NextJS routing [wasn't yet supported](https://github.com/algolia/instantsearch/issues/5241#issuecomment-1451970254).

Example AjMadison page using Autocomplete and InstantSearch Hooks: https://www.ajmadison.com/c/cooking-ranges/

**Comment by Haroenv** [comment]
This is really interesting! I believe for your feedback on conditional widgets being complicated we're currently experimenting with a new method of instantiating widgets that makes this tradeoff less notable, especially during server side rendering.
